### ISSUE TITLE
[Feature Request]: Allow handler in app.on("open-url", handler) to choose whether to activate the app or not

### ISSUE THREAD

**Issue by safris** [electron/electron#42051]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Problem Description

Consider a Notification with an action such as "Delete", to indicate to the app that the subject message of the Notification should be deleted.

With an implementation that uses registered protocols, the "Delete" action is delivered to Electron via the app.on("open-url", handler) listener.

When Electron receives an "open-url" event, it is automatically activated (i.e. gets focus, and is made frontmost), which is not necessarily desired (such as when the user wishes to "Delete" and thus ignore the message).

### Proposed Solution

When Electron receives an "open-url" event, _do not_ automatically activate the app (i.e. focusing it and making it frontmost).

To activate the app, `app.focus({ steal: true })` can be called manually.

Since this would be a breaking change of the behavior of the "open-url" event handler, I propose implementation of a new method on `app` to "switch on or off" this new behavior. As in, something akin:
```javascript
app.dontAutoFocusAppOnOpenUrl(true);
```

### Alternatives Considered

The problem can be abstracted as follows: "How can Electron receive a message from another application or the OS, and handle that message without automatically being activated?"

Listening for second instance invocations of the app from the OS with `app.on("second-instance", handler)` provides a solution to this problem. However, this approach would require the invoker to construct a CLI-based command to spawn a new shell, in which a new instance of the app is spawned, only to deliver a message payload to the running instance's `app.on("second-instance", handler)` handler.

Another solution is via an embedded web server running as a utility process. This approach, however, is not feasible due a couple edge cases, one of which is: the need to register a listening port that does not persist upon app restart.

**Comment by rmartins90** [comment]
@safris did you find a workaround for this?

**Comment by safris** [comment]
@rmartins90, my workaround was to use the method mentioned in the **Alternatives Considered** section of the description (via `app.on("second-instance", handler)`). This workaround forfeits the ease of use of the `"open-url"` API in exchange for the control of app activation. The `"second-instance"` approach is cumbersome, as you become responsible for:
1. Ensuring the caller knows the path of the app.
2. Ensuring the arguments are encoded in a bug-free sort of way (this depends on the kind of data you desire to pass to the app).

**Comment by rmartins90** [comment]
Thank you @safris. But that doesn't work on macOS with deep links, right? Or am I missing something?

**Comment by safris** [comment]
@rmartins90, correct. The workaround I provided will only help the use-case involving the need to dispatch some information to a running (or not running) Electron application from another application, where the other application allows one to issue a `system` command. If your use-case involves clicking on an URL in MacOS, or calling `open <URL>` from the Terminal, the workaround I provided will not help you.
